packages/blade/src/components/BottomSheet/_decisions/decisions.md

```diff
@@ -0,0 +1,276 @@
+  - [Accessibility](#accessibility)
+  - [Open questions](#open-questions)
+  - [References](#references)
+
+## Design
+
+- [Figma - BottomSheet](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=26477%3A578228&t=1RMqjs99P33Udj0C-0)
+
+## Anatomy
+
+<img src="./bottomsheet-anatomy.png" alt="BottomSheet Anatomy" width="100%" />
+
+
+Components:
+
+- BottomSheet
+  - BottomSheetHeader
+    - BottomSheetHeaderLeading
+    - BottomSheetHeaderTrailing
+  - BottomSheetBody
+  - BottomSheetFooter
+    - BottomSheetFooterLeading
+    - BottomSheetFooterTrailing
+
+## API
+
+Sample usage:
+
+```jsx
+import { BottomSheet } from '@razorpay/blade';
+
+<BottomSheet open={boolean} snapPoints={[]}>
+  <BottomSheetHeader>
+    <BottomSheetHeaderLeading title="Payments Links" prefix={BladeIcon} />
+    <BottomSheetHeaderTrailing visual={Link | Action | Icon} />
+  </BottomSheetHeader>
+  <BottomSheetBody>
+    <Text>Body Content<Text>
+  </BottomSheetBody>
+  <BottomSheetFooter>
+    <BottomSheetFooterLeading title="Footer Title" prefix={icon} />
+    <BottomSheetFooterTrailing
+      actions={{
+        primaryAction: { text: 'Know more', onClick: () => {} },
+        secondaryAction: { text: 'Read Docs', onClick: () => {} },
+      }}
+    />
+  </BottomSheetFooter>
+</BottomSheet>;
+```
+
+**Design Constraints**
+
+- BottomSheetHeader:
+  - Leading:
+    - Title
+    - Prefix
+  - Trailing:
+    - empty
+    - icon
+    - link
+    - action
+
+- BottomSheetFooter:
+  - Leading
+    - title
+    - prefix
+  - Trailing
+    - Primary action button
+    - Secondary action button
+
+- The BottomSheetFooter will behave similar to Card's footer.
+- The Header, Footer can be individually omitted
+
+### `BottomSheet`
+
+We'll expose a `BottomSheet` component with the following API:
+
+| Prop            | Type        | Default                 | Description                                                                                     | Required |
+| --------------- | ----------- | ----------------------- | ----------------------------------------------------------------------------------------------- | -------- |
+| open            | `boolean`   | `false`                 | toggles bottom sheet state content                                                              |          |
+| snapPoints      | `[number, number, number]`  | `[0.35, 0.5, 0.85]` | Snappoints in which the bottom sheeet will rest on, this accepts a number between 0 & 1 which maps to the total view height of the screen. 0.5 means 50% of screen height.                                              |          |
+| onDismiss       | `Callback`  | `undefined`             | called when the bottom sheet is closed, either by user state, hitting `esc` or tapping backdrop |          |
+| initialFocusRef | `React.Ref` | `undefined`             | ref element you want to get keyboard focus when opening the sheet                               |          |
```

Comment:
Does this mean that if I pass a ref of a TextInput, it will get keyboard focus and I can type directly? Would it also mean that the ref we receive from the user needs to be focusable?

Reply:
> Does this mean that if I pass a ref of a TextInput, it will get keyboard focus and I can type directly

Yes initially it will be focused, check this [ariakit example](https://ariakit.org/examples/dialog-react-router) the textarea is focused initially when the modal opens. 

> Would it also mean that the ref we receive from the user needs to be focusable?

Yes the element needs to be "any [tababble](https://allyjs.io/data-tables/focusable.html) element"

Reply:
Got it. Do we have a way to enforce consumers to only pass tabbable element? Or will we handle that via documentation only?

Reply:
> Or will we handle that via documentation only?

This should be fine. 

Tho even if they pass a non-tabbable element, internally we will call `focus` on the element, so if the focus method is not found or it's not a focusable element it will be a noop.